Dockerfile

```diff
@@ -70,142 +78,121 @@ RUN sudo apt install -y --no-install-recommends \
-# RUN sudo apt install python-rosdep python-rosinstall python-rosinstall-generator python-wstool build-essential -y --no-install recommends
-
-
-# ## AirSim Installation Dependencies (TODO: necessary?)
-# # # RUN sudo apt-get install python3 python3-pip python3-dev sudo libglu1-mesa-dev xdg-user-dirs pulseaudio -y --no-install-recommends
-# # RUN sudo apt install python3 python3-pip python3-dev sudo libglu1-mesa-dev xdg-user-dirs -y --no-install-recommends
-# # RUN sudo apt install build-essential cmake cppcheck gdb git vim wget tmux less htop python python-pip python-tk -y --no-install-recommends
-
-# Cleanup
-RUN sudo apt clean autoremove
-
-# # ## Python (TODO: necessary?)
-# # RUN pip3 install setuptools wheel --no-cache-dir
-# # RUN pip3 install 'ue4cli>=0.0.41' 'conan-ue4cli>=0.0.21' ue4-ci-helpers --no-cache-dir
-# # RUN pip3 install msgpack-rpc-python numpy airsim --no-cache-dir
-# # RUN pip install colored-traceback catkin_tools msgpack-rpc-python torch future --no-cache-dir
-
-RUN sudo chown -R $USERNAME /home/$USERNAME
-## Unreal Setup (https://bitbucket.org/castacks/cluster/wiki/airsim_ros)
-# ( https://microsoft.github.io/AirSim/build_linux/)
+RUN git clone -b 4.25 https://$GITHUB_USER:$GITHUB_PWD@github.com/EpicGames/UnrealEngine.git
 WORKDIR /$FOLDER_NAME/UnrealEngine
-# RUN sudo chown -R $USERNAME /$FOLDER_NAME
 RUN ./Setup.sh
 RUN ./GenerateProjectFiles.sh
 USER $USERNAME
 RUN make
-WORKDIR /$FOLDER_NAME
 
-## AirSim Setup (https://microsoft.github.io/AirSim/build_linux/)
-WORKDIR AirSim
+
+# =======================================================================
+#  AirSim v1.4.0-linux (https://microsoft.github.io/AirSim/build_linux/)
+# =======================================================================
+WORKDIR /$FOLDER_NAME
+RUN git clone --depth 1 -b v1.4.0-linux https://github.com/Microsoft/AirSim.git
+WORKDIR /$FOLDER_NAME/AirSim
 RUN ./setup.sh
 RUN ./build.sh
 # use ./build.sh --debug to build in debug mode
+# TODO: Copy Plugins folder to host if possible
 
 
-## Mapping Code (https://github.com/Amigoshan/tartanair.git)
-ARG WORKSPACE=catkin_ws
+# ==========================
+# ROS Melodic
+# ==========================
+# TODO: Still have problem on this installation. It works after trying to add some more dependencies but the list is not firmed.
+RUN sudo sh -c 'echo "deb http://packages.ros.org/ros/ubuntu bionic main" > /etc/apt/sources.list.d/ros-latest.list'
+RUN curl -sSL 'http://keyserver.ubuntu.com/pks/lookup?op=get&search=0xC1CF6E31E6BADE8868B172B4F42ED6FBAB17C654' | sudo apt-key add -
+WORKDIR /home/$USERNAME
+# Pre-requisites for ros-melodic-desktop-full
+RUN sudo apt-get update && sudo apt-get install ros-melodic-perception ros-melodic-simulators ros-melodic-urdf-sim-tutorial ros-melodic-perception-pcl ros-melodic-pcl-conversions ros-melodic-pcl-ros libgazebo9-dev ros-melodic-gazebo-dev ros-melodic-gazebo-plugins ros-melodic-gazebo-ros ros-melodic-gazebo-ros-pkgs -y --no-install-recommends
```

Comment:
Did this instruction work for you? I put this in because last week I needed to do this. But recently Melvin and I both built, this instruction didn't work for us and we had to revert to RUN sudo apt-get install ros-melodic-desktop-full

Reply:
Either didn't work for me when I build from clean state, ie. remove all docker images, containers. What I tried was when I hit dependencies issue, I try to add one dependency at a time and try again. Seems like it works after adding some (not all) dependencies. 

Abit weird, need to troubleshoot further though.

Reply:
Tested again and it works.

Reply:
Done!